fraud/proof.go

```diff
@@ -80,3 +93,39 @@ func Unmarshal(proofType ProofType, msg []byte) (Proof, error) {
 	}
 	return unmarshaler(msg)
 }
+
+// MonitorProofs enables metrics to monitor fraud proofs.
+func MonitorProofs(proofTypes ...ProofType) func(Getter) {
+	return func(store Getter) {
+		for _, proofType := range proofTypes {
+			counter, _ := meter.AsyncInt64().Counter(string(proofType),
+				instrument.WithUnit(unit.Dimensionless),
+				instrument.WithDescription("Stored fraud proof"),
+			)
+			err := meter.RegisterCallback(
+				[]instrument.Asynchronous{
+					counter,
+				},
+				func(ctx context.Context) {
+					proofs, err := store.Get(ctx, proofType)
+					switch err {
+					case nil:
+						counter.Observe(ctx,
+							int64(len(proofs)),
+							attribute.String("proofType", string(proofType)),
+						)
+					case datastore.ErrNotFound:
+						counter.Observe(ctx, 0)
+						return
+					default:
+						counter.Observe(ctx, 0, attribute.String("err", err.Error()))
+					}
```

Comment:
I think Counter observes increasing increment, so there should be no reason to observe 0 increment since it won't affect the metric.

Also setting arbitrary error as vector value could lead to problems in metric storage indexes. I would suggest mapping error into error codes or simply storing as bool(err != nil). This way amount of possible values in vector is limited.